plutus-core/plutus-core/src/PlutusCore/Builtin/KnownType.hs

```diff
@@ -361,13 +358,6 @@ makeKnownOrFail x = case makeKnown x of
     BuiltinFailure _ _           -> EvaluationFailure
 {-# INLINE makeKnownOrFail #-}
 
--- | Same as 'readKnown', but the cause of a potential failure is the provided term itself.
-readKnownSelf
-    :: (ReadKnown val a, AsUnliftingEvaluationError err, AsEvaluationFailure err)
-    => val -> Either (ErrorWithCause err val) a
-readKnownSelf val = fromRightM (throwBuiltinErrorWithCause val) $ readKnown val
-{-# INLINE readKnownSelf #-}
-
```

Comment:
Is this change related to the task or is it an opportunistic improvement?

Reply:
It's not possible to define this function without prismatic error type constraint unless I have a copy of this function per error type. 

Instead, I removed this function entirely and handled error when `readKnown` is used where error type is actually realized

Reply:
Gotcha! Thanks for the clarification.